docs/sdk-and-tools/sdk-nestjs/sdk-nestjs-monitoring.md

```diff
@@ -0,0 +1,197 @@
+
+```typescript
+import { CpuProfiler } from '@multiversx/sdk-nestjs-monitoring';
+
+const httpReqCpuProfiler = new CpuProfiler();
+await doHttpRequest();
+httpReqCpuProfiler.stop(`doHttpRequest() execution time`);
+
+const cpuProfiler = new CpuProfiler();
+await doSomethingCpuIntensive();
+cpuProfiler.stop(`doSomethingCpuIntensive() execution time`);
+```
+The output of the code above will be <br/>
+
+`doHttpRequest() execution time: 100ms, CPU time: 1ms`
+`doSomethingCpuIntensive() execution time: 20ms, CPU time 18ms`
+
+*Note that a big execution time does not necessarily have an impact on the CPU load of the application. That means that, for example, while waiting for an HTTP request, the JavaScript thread can process other things. That is not the case for CPU time. When a method consumes a lot of CPU time, Javascript will not be able to process other tasks, potentially causing a freeze until the CPU-intensive task is complete.*
+
+---
+
+[comment]: # (mx-context-auto)
+
+## Interceptors
+The package provides a series of [Nestjs Interceptors](https://docs.nestjs.com/interceptors) which will automatically log and set the CPU and overall duration for each request in a prom histogram ready to be scrapped by Prometheus.
+
+`LoggingInterceptor` interceptor will set the execution time of each request in a Prometheus histogram using [performance profilers](#performance-profiler).
+
+`RequestCpuTimeInterceptor` interceptor will set the CPU execution time of each request in a Prometheus histogram using [cpu profiler](#cpu-profiler).
+
+*Both interceptors expect an instance of `metricsService` class as an argument.*
+
+```typescript
+import { MetricsService, RequestCpuTimeInterceptor, LoggingInterceptor } from '@multiversx/sdk-nestjs-monitoring';
+
+async function bootstrap() {
+  // AppModule imports MetricsModule
+  const publicApp = await NestFactory.create(AppModule);
+  const metricsService = publicApp.get<MetricsService>(MetricsService);
+
+  const globalInterceptors = [];
+  globalInterceptors.push(new RequestCpuTimeInterceptor(metricsService));
+  globalInterceptors.push(new LoggingInterceptor(metricsService));
+
+  publicApp.useGlobalInterceptors(...globalInterceptors);
+}
+```
+
+[comment]: # (mx-context-auto)
+
+## MetricsModule and MetricsService
+
+`MetricsModule` is a [Nestjs Module](https://docs.nestjs.com/modules) responsible for aggregating metrics data through `MetricsService` and exposing them to be consumed by Prometheus. `MetricsService` is extensible, you can define and aggregate your own metrics and expose them. By default it exposes a set of metrics created by the interceptors specified [here](#interceptors). Most of the Multiversx packages expose metrics by default through this service. For example [@multiversx/sdk-nestjs-redis](https://www.npmjs.com/package/@multiversx/sdk-nestjs-redis) automatically tracks the execution time of each redis query, overall redis health and much more, by leveraging the `MetricsService`.
```

Comment:
"Most of the Multiversx packages" -> typo X.

Reply:
done